    Individual Plan


github link:https://github.com/ting20231227-ctrl/dsci100-individual

    (1).Importing data


In [3]:
library(tidyverse)
url='https://drive.google.com/uc?export=download&id=1DPK-faIDmjTfkVzy1O-nmrd3lmK97dmX'
players<-read_csv(url)
head(players)

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors
Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


experience subscribe
1 Pro         TRUE    
2 Veteran     TRUE    
3 Veteran    FALSE    
4 Amateur     TRUE    
5 Regular     TRUE    
6 Amateur     TRUE    
  hashedEmail                                                      played_hours
1 f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d 30.3        
2 f3c813577c458ba0dfef80996f8f32c93b6e8af1fa939732842f2312358a88e9  3.8        
3 b674dd7ee0d24096d1c019615ce4d12b20fcbff12d79d3c5a9d2118eb7ccbb28  0.0        
4 23fe711e0e3b77f1da7aa221ab1192afe21648d47d2b4fa7a5a659ff443a0eb5  0.7        
5 7dc01f10bf20671ecfccdac23812b1b415acd42c2147cb0af4d48fcce2420f3e  0.1        
6 f58aad5996a435f16b0284a3b267f973f9af99e7a89bee0430055a44fa92f977  0.0        
  name      gender Age
1 Morgan    Male    9 
2 Christian Male   17 
3 Blake     Male   17 
4 Flora     Female 21 
5 Kylie     Male   21 
6 Adrian    Female 17

Data description: I downloaded the players information, which shows there are 196 players in total and 7 variables.
Experience, categorical variable, the level of experience of players in the game.
Subscribe, logical variable, whether players subscribe to the game news letter or not.
hashedEmail, string, an anonymized email that targeted different players.
played_hours, numeric variable, the time in hours that players played in this game.
name, string, player's name.
gender, categorical variable, player's gender.
Age, numeric variable, player's age in years.

In [6]:
library(tidyverse)
url='https://drive.google.com/uc?export=download&id=1T3Vgo6ee6eJ40PTWLs8PKS_FU_0borDG'
sessions<-read_csv(url)
head(sessions)

Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


hashedEmail                                                     
1 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
2 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
3 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
4 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
5 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
6 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
  start_time       end_time         original_start_time original_end_time
1 30/06/2024 18:12 30/06/2024 18:24 1.71977e+12         1.71977e+12      
2 17/06/2024 23:33 17/06/2024 23:46 1.71867e+12         1.71867e+12      
3 25/07/2024 17:34 25/07/2024 17:57 1.72193e+12         1.72193e+12      
4 25/07/2024 03:22 25/07/2024 03:58 1.72188e+12         1.72188e+12      
5 25/05/2024 16:01 25/05/2024 16:12 1.71665e+12         1.71665e+12      
6 23/06/2024 15:08 23/06/2024 17:10 1.71916e+12         1.71916e+12

This data contains the information of player's game play session, it contains 1535 sessions and 5 variables. Each of the row corresponds to a player.
-Hashed email, string variable, an anonymized identifier linking each sessions to players in the 'players_csv'.
-start_time, string variable, timestamp indicating when the session time started.
-end_time, string variable, timestamp indicating when the session time ended.
-original_start_time, numeric variable, the Unix timestamp, which is equivalent to the session start time.
*original_end_time, numeric variable, the Unix timestamp, which is equivalent to the session end time.

    （2）. State question 3, we are interested in demand forecasting, what time is windows are most likely to have large number of simultaneous players.

The specific question is "Can the usage in hour be the predictor of the number of predicting the simultaneous players?"

In [10]:
library(lubridate)
sessions_clean <- sessions |>
  mutate(
    start_time = dmy_hm(start_time),   
    hour = hour(start_time)            
  )
sessions_clean

hashedEmail                                                     
1    bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
2    36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
3    f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
4    bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
5    36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
6    bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
7    fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
8    ad6390295640af1ed0e45ffc58a53b2d9074b0eea694b16210addd44d7c81f83
9    96e190b0bf3923cd8d349eee467c09d1130af143335779251492eb4c2c058a5f
10   36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
11   fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
12   f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
13   f2826fb8dbce4d450348f99cb27ade184b713998d9679780442efaaf218038f2
14   b622593d2ef8b337dc554acb307d04a88114f2bf453b18fb5d2c80052aeb2319
15   24d4892c2c3ce11d3e54c3bf31ee218901cac7a2f564911d67424c8fdf0874fe
16   fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
17   88247d9a46fc214a12485dcbcbb03a8ddebfe8c1ec5fe209cd660147625c8e62
18   f6daba428a5e19a3d47574858c13550499be23603422e6a0ee9728f8b53e192d
19   36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
20   fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
21   bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
22   36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
23   fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
24   ad6390295640af1ed0e45ffc58a53b2d9074b0eea694b16210addd44d7c81f83
25   ad6390295640af1ed0e45ffc58a53b2d9074b0eea694b16210addd44d7c81f83
26   fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
27   36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
28   bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
29   b622593d2ef8b337dc554acb307d04a88114f2bf453b18fb5d2c80052aeb2319
30   fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
⋮    ⋮                                                               
1506 ad6390295640af1ed0e45ffc58a53b2d9074b0eea694b16210addd44d7c81f83
1507 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
1508 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
1509 b3510c708bd50bf9f75e6e02bb6fe14edb705e0ea671ee769a45c613750e8b58
1510 bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
1511 e74c60a92c0100e7240be56d66969db85856152b048c639331a3595f901461ae
1512 b622593d2ef8b337dc554acb307d04a88114f2bf453b18fb5d2c80052aeb2319
1513 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
1514 e74c60a92c0100e7240be56d66969db85856152b048c639331a3595f901461ae
1515 fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
1516 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
1517 a1e0438d568b9c05a93cadaf99969e4d44794d19dea31edd723cbc916f878c33
1518 fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
1519 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
1520 55d24216db39c27e1f17cc43d3127cbf8ed76ada6d098202b53ded319855e2c1
1521 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
1522 b622593d2ef8b337dc554acb307d04a88114f2bf453b18fb5d2c80052aeb2319
1523 b622593d2ef8b337dc554acb307d04a88114f2bf453b18fb5d2c80052aeb2319
1524 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
1525 fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
1526 18936844e06b6c7871dce06384e2d142dd86756941641ef39cf40a9967ea14e3
1527 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
1528 18936844e06b6c7871dce06384e2d142dd86756941641ef39cf40a9967ea14e3
1529 f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
1530 b622593d2ef8b337dc554acb307d04a88114f2bf453b18fb5d2c80052aeb2319
1531 36d9cbb4c6

In [11]:
hourly_usage <- sessions_clean |>
  group_by(hour) |>
  summarize(session_count = n())
hourly_usage

hour session_count
1   0   128          
2   1    79          
3   2   152          
4   3   131          
5   4   151          
6   5    88          
7   6    74          
8   7    56          
9   8    12          
10  9    13          
11 10     3          
12 12     1          
13 13     2          
14 14    14          
15 15    49          
16 16    36          
17 17    48          
18 18    64          
19 19    58          
20 20    74          
21 21    91          
22 22    89          
23 23   122